### DISCUSSION TITLE
Cannot do a cross / outer apply

### DISCUSSION THREAD

**Discussion by NicolasLOJO** [nhibernate/nhibernate-core#2956] [Q&A]
Hello,
I cannot find any documentation on how to do some cross / outer apply with NHibernate and i cannot do a named query because all my query is dynamicly created.

This is a exemple on what i want to do with NHibernate :
`CROSS apply
(
    VALUES
        (foo.a, foo.b, foo.c, foo.d, foo.e),
        (bar.a, bar.b, bar.c, bar.d, bar.e)
) AS toto (a, b, c, d, e)`

It is possible to do this ? or not integrated to NHibernate ?

**Comment by fredericDelaporte** [comment]
Named queries are not the only way to execute SQL queries with NHibernate. Use a SQL query. See [here](https://nhibernate.info/doc/nhibernate-reference/querysql.html#querysql-creating).
